contained-innersource.md

```diff
@@ -0,0 +1,70 @@
+## Title  
+
+Contained InnerSource
+
+## Patlet
+
+Apply InnerSource methods to facilitate collaboration in a cross divisional 
+project but don't invest in soliciting contributions from outside of that
+project.
```

Comment:
How is this different from traditional walled-off source?

Reply:
Isn't this more of a "step toward" InnerSource, rather than a "pattern of" InnerSource? Do we make such distinctions? I worry that if a pattern like this is adopted, it can be seen as "good enough" by supporters of the status-quo and work against folks that want more internally open patterns.

Reply:
> Isn't this more of a "step toward" InnerSource, rather than a "pattern of" InnerSource?

I think so.  To be fair, I think that the *Resulting Context* calls that out clearly.

> Do we make such distinctions?

We haven't learned how to be that formal about it, yet.

> it can be seen as "good enough"

That's true and a good point.  @NewMexicoKid is there a place in the pattern format to put caveats or known "gotchas" with a particular pattern?

Reply:
In my interpretation of this pattern **Contained InnerSource** can indeed be seen as a "step towards" InnerSource. However I don't see this as a problem, as a step is better than no step. And if the situation really doesn't allow for more, I would take it :)

One other angle to look at this is to consider the  **Contained InnerSource** idea like a pilot project where the concept can be tried out. One hope is that the developers that experience it may advocate for it more strongly afterwards, and if they do they can use the specific experiences that they have made as examples that the decision makers can more easily relate to.

Reply:
> That's true and a good point. @NewMexicoKid is there a place in the pattern format to put caveats or known "gotchas" with a particular pattern?

Those go into the Resulting Context section; and it is not uncommon for significant "gotchas" or issues identified there to point to further patterns that could then be applied.